bin/run-circleci-tests

```diff
@@ -4,6 +4,18 @@ set -e
 
 make install_system_dependencies
 
+# Ensure Redis is up and running before starting to run any tests:
+# In the past, individual tests from our unit test suite
+# would sometimes fail with a "ConnectionRefusedError"
+# caused by Redis not starting for the container running the tests.
+# The solution that we use below comes from the following support thread:
+# https://discuss.circleci.com/t/redis-server-problems-unable-to-connect-localhost-6379/4040
+until bash -c "sudo service redis-server status"; do
```

Comment:
After some reading it turns out that circle times out work only for each command (I didn't find any mention of build timeout per se). Nevertheless timing out after 4 minutes of waiting for redis won't hurt.